src/components/EditFeatureDrawer/EditFeatureGeometryToolbar/index.tsx

```diff
@@ -0,0 +1,283 @@
+      };
+      setRevision(r => r + 1);
+    }
+  };
+
+  const redoEdit = () => {
+
+    const editSource = editLayer?.getSource();
+    const features = editSource?.getFeatures();
+
+    if (!features) {
+      return;
+    }
+
+    const clone = gjFormat.writeFeaturesObject(features);
+    const nextSource = editHistory.current.future[editHistory.current.future.length - 1];
+    if (!_isEmpty(nextSource)) {
+      editSource?.clear();
+      editSource?.addFeatures(gjFormat.readFeatures(nextSource));
+
+      editHistory.current = {
+        future: editHistory.current.future.slice(0, -1),
+        past: [...editHistory.current.past, clone].slice(-historyLength)
+      };
+      setRevision(r => r + 1);
+    }
+  };
+
+  const onDrawEnd = (e: OlDrawEvent) => {
+    updateRevision();
+    if (!feature.geometry.type.toLocaleLowerCase().startsWith('multi')) {
+      // replace the existing geometry by the new one
+      editLayer?.getSource()?.clear();
+    }
+  };
+
+  const updateRevision = () => {
+
+    const features = editLayer?.getSource()?.getFeatures();
+
+    if (!features) {
+      return;
+    }
+
+    const clone: FeatureCollection = gjFormat.writeFeaturesObject(features);
+    editHistory.current = {
+      ...editHistory.current,
+      past: [...editHistory.current.past, clone].slice(-10)
+    };
+    setRevision(r => r + 1);
+  };
+
+  const btnTooltipProps = {
+    tooltipPlacement: 'left' as TooltipPlacement,
+    tooltipProps: {
+      mouseEnterDelay: 0.5
+    }
+  };
+
+  if (!editLayer) {
+    return <></>;
+  }
+
+  return (
+    <Toolbar
+      className="geometry-edit-tb"
+      alignment="vertical"
+    >
+      <ToggleGroup>
+        {
+          allowedEditMode.includes('CREATE') ?
+            <DrawButton
+              icon={
+                <FontAwesomeIcon icon={faPencil} />
+              }
+              pressedIcon={
+                <FontAwesomeIcon icon={faPencil} />
+              }
+              name="draw"
+              digitizeLayer={editLayer}
+              tooltip={t('EditFeatureGeometryToolbar.draw')}
+              drawType={feature.geometry.type as DrawType}
+              onDrawEnd={onDrawEnd}
+              {...btnTooltipProps}
+            />
+            : <></>
+        }
+        {
+          allowedEditMode.includes('UPDATE') ?
+            <ModifyButton
+              icon={
+                <FontAwesomeIcon icon={faDrawPolygon} />
+              }
+              pressedIcon={
+                <FontAwesomeIcon icon={faDrawPolygon} />
+              }
+              name="edit"
+              digitizeLayer={editLayer}
+              tooltip={t('EditFeatureGeometryToolbar.edit')}
+              onModifyStart={updateRevision}
+              onModifyEnd={updateRevision}
+              onTranslateEnd={updateRevision}
+              {...btnTooltipProps}
+            />
+            : <></>
+        }
+        {
+          allowedEditMode.includes('DELETE') ?
```

Comment:
This is actually refers to the deletion of the _geometry_ not the _feature_. Is this intended?

Reply:
Yes, the requirement was that in this case both form editing and geometry editing tools should be restricted according to the configuration.

Reply:
Alright, thanks!